### PR TITLE
AMM amplification params in constructor

### PR DESCRIPTION
**Pull request by RedFox20**
- [x] Add new wrapper for `setAmplificationParameters`
- [x] Fix test failures

### CODE REVIEW

**Comment by RedFox20** [comment]
- [ ] Add new wrapper for `setAmplificationParameters`
- [ ] Fix test failures

File: contracts/amm/TempusAMM.sol [hunk 1]
```diff
@@ -678,15 +675,34 @@ contract TempusAMM is BaseMinimalSwapInfoPool, StableMath, IRateProvider {
...
+     *
+     * NOTE: Internally, the amplification parameter is represented using higher precision. The values returned by
+     * `getAmplificationParameter` have to be corrected to account for this when comparing to `rawEndValue`.
+     */
+    function setAmplificationParameters(
+        uint256 rawStartValue,
+        uint256 rawEndValue,
+        uint256 startTime,
+        uint256 endTime
+    ) public {
```
Comment (yuval-weiss) [inline contracts/amm/TempusAMM.sol:696]: This is needs an `authenticate` modifier

File: contracts/amm/TempusAMM.sol [hunk 2]
```diff
@@ -140,8 +138,7 @@ contract TempusAMM is BaseMinimalSwapInfoPool, StableMath, IRateProvider {
         _scalingFactor0 = _computeScalingFactor(IERC20(address(token0)));
         _scalingFactor1 = _computeScalingFactor(IERC20(address(token1)));
 
-        uint256 initialAmp = Math.mul(amplificationParameter, _AMP_PRECISION);
-        _setAmplificationData(initialAmp);
+        setAmplificationParameters(amplificationStart, amplificationEnd, pool.startTime(), pool.maturityTime());
```
Comment (yuval-weiss) [inline contracts/amm/TempusAMM.sol:141]: I think we should `require` that `amplificationStart < amplificationEnd`. If this condition is false it means we didn't use proper params for deploying the amm...

File: contracts/amm/TempusAMMFactory.sol [hunk 3]
```diff
@@ -1,60 +0,0 @@
-// SPDX-License-Identifier: GPL-3.0-or-later
```
Comment (mijovic) [inline contracts/amm/TempusAMMFactory.sol:1]: Finally deleted

File: test/amm/TempusAMM.test.ts [hunk 4]
```diff
@@ -55,7 +55,8 @@ describeForEachPool("TempusAMM", (testFixture:PoolTestFixture) =>
     if (params.ammBalanceYield != undefined && params.ammBalancePrincipal != undefined) {
       await tempusAMM.provideLiquidity(owner, params.ammBalancePrincipal, params.ammBalanceYield, TempusAMMJoinKind.INIT);
     }
-    if (params.amplifyEnd != undefined) {
+    if (params.amplifyStart != params.amplifyEnd) {
```
Comment (mijovic) [inline test/amm/TempusAMM.test.ts:58]: perfect

**Review by mijovic — APPROVED** [review]

**Review by mijovic — APPROVED** [review]

**Pull request merged**